### PR TITLE
[feat] Hiding Markers when Search and Filters are Activated

### PR DESCRIPTION
**Pull request by nehaahussain**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- pass in filteredProjects from MapViewScreen component to AddMarkers.
- created 2 functions, one hiding a specific marker and one showing it on the map.
- re-render clusters on the map when searching a project
- created `markerMap` to map the project ids to its corresponding marker. 

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

https://github.com/user-attachments/assets/080385ba-8f89-4819-bc84-f11fd4f3f59c

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- type a project name in the search bar

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."
- `google.maps.Marker` vs `google.maps.marker.AdvancedMarkerElement` -> `setMap()` is part of Marker not AdvancedMarkerElement (check comment)

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: api/maps/MarkerInfoWindow.tsx [hunk 1]
```diff
@@ -87,8 +89,16 @@ export const MarkerInfoWindow = ({
   useEffect(() => {
     if (marker && clusterer) {
       clusterer.addMarker(marker);
+      markerMap.set(projectId, marker as unknown as google.maps.Marker);
```
Comment (nehaahussain) [inline api/maps/MarkerInfoWindow.tsx:92]: this bypasses the type-checking system but may cause runtime issues bc of compatibility. `marker` = AdvancedMarkerElement but `markerMap` takes in google.maps.Marker.

Comment (itsliterallymonique) [reply api/maps/MarkerInfoWindow.tsx:92]: hmmm, I don't want to run the risk of runtime errors and when I checked it on my VSCode it did say Marker is depreciated. Since this feature does not need to be merged in ASAP, let's work together on a fix after Thanksgiving Break.

File: api/maps/AddMarkers.tsx [hunk 2]
```diff
@@ -129,6 +130,50 @@ export default function AddMarker({
     return setClusterer;
   }, [map]);
 
+  const markerMap = useRef<Map<number, google.maps.Marker>>(new Map());
```
[the remark this answers is not in the document]
Comment (nehaahussain) [reply api/maps/AddMarkers.tsx:133]: This creates `markerMap` which is like a hashmap that maps the project ids to its corresponding marker (google.maps.Marker). `useRef` is just a hook that makes it so that the component doesn't re-render every time we update the map. Instead, `useRef` creates a mutable object (markerMap.current) that exists across re-renders of the component.
